Migrate kubernetes-incubator/external-storage/nfs

kmova opened the pull request:
This PR covers the migration of code from https://github.com/kubernetes-incubator/external-storage/tree/master/nfs to this repo. 

The tasks include. 
- [x] import nfs from external-storage repo (git filter-branch --subdirectory-filter nfs -- --all)
- [x] update README.md (3efa64d7e33cc5cf711f1086a8cc630e731a1dae)
- [x] update go import path (eb12c1db9298878cef5b5dad3d0290997fe647e1)
- [x] update with image name and registry location (a1c4fd8d7da4e6da012ecbf45d2a19fbf450d9ad)
- [x] update vendor (0140304ce7f3cbe220c90484f5e235ab1ed46dd2)

With the changes made in this PR, the following build commands work. 
- `make container` - generates `quay.io/external_storage/nfs-ganesha-server-and-provisioner:latest`
- `make clean`

Tasks to be done in future PRs: 
- Setup the new quay registry location
- Refactor E2e
- Setup build automation (PROW) to push containers to registry
- Update the docs/deployment.md to reflect new/updated testing scenarios.

Commit:
Move nfs things into nfs directory

Commit:
Move e2e into test

Commit:
Move demos-hostpath to top-level docs, nfs to docs/demo

Commit:
Fix import statements

Commit:
Manually vendor self in

Commit:
Fix e2e test hardcoded path

Commit:
Rename kube-config dir -> kubernetes

Commit:
Separate authorization (psp,scc,rbac) doc from deployment

Commit:
Verify golint, vet, & boilerplate

Commit:
Fix boilerplate, golint, govet

Commit:
Verify library controller code

Commit:
Update import statements and (most) instances of nfs-provisioner->external-storage

Commit:
Fix nfs-provisioner build by updating vendor

Commit:
Update READMEs

Commit:
Fix nfs build

Commit:
Update demos (hostpath), docs

wongma7 committed:
Update README.md

Commit:
Use shared top-level vendor directory

Commit:
Fix verify golint

Commit:
Release nfs provisioner v1.0.4

Commit:
Add nfs changelog and efs+nfs release inhfo

wongma7 committed:
Update deployment.md

Commit:
setgid on efs and nfs

Commit:
Update all provisioners' imports, glide.yaml

Commit:
Update lib controller and all provisioners except NFS tests

Commit:
Vendor kube e2e framework for nfs, separate from top-level

Commit:
Release nfs v1.0.5

Commit:
Move DefaultLeaseDuration... parameters to controller pkg and flesh out code doc comments

Commit:
Remove non-emptydir nfs provisioner pod option

Commit:
Add 'list events' to openshift clusterrole examples

Commit:
Do clean up in NFS Dockerfile for smaller image

Commit:
Release nfs-provisioner v1.0.6

Commit:
Set number of files resource limit before starting ganesha

wongma7 committed:
Merge pull request #72 from wongma7/ulimit

Set number of files resource limit before starting ganesha

Commit:
Update docs and samples to include CAP_SYS_RESOURCE

Commit:
Release v1.0.7

Commit:
Update all provisioners to use new NewProvisionController

Commit:
Make createVolume return struct instead of ~7 values

Commit:
Add mountOptions StorageClass parameter

wongma7 committed:
Merge pull request #84 from wongma7/mountoptions

Add mountOptions StorageClass parameter for NFS

Commit:
Replace root-squash argument with a rootSquash SC parameter

Commit:
Put ganesha logs in /export

Commit:
Watch for unexpected stop of ganesha.nfsd and restart if seen

Commit:
Release nfs-provisioner v1.0.8

wongma7 committed:
Merge pull request #100 from wongma7/v1.0.8

Release nfs-provisioner v1.0.8

Commit:
Log nfs-provisioner pod logs

Commit:
Remove provisioner-specific verification

Commit:
Fix all gofmt, gometalinter, govet errors except in local-volume

Commit:
Add event list permission to ClusterRole

Fixes: #152

Commit:
Create top-level Makefile and use it in travis CI

wongma7 committed:
Merge pull request #160 from wongma7/makefile

Create top-level Makefile and use it in travis CI

Commit:
Standardize efs, glusterblock, nfs makefiles

kokhang committed:
Incorporate K8S API changes.
- Importing API types from k8s.io/api
- Update client-go and apimachinery version
- Remove lib/helper and use the kubernetes one temporary

samuelmanzer committed:
Fix daemon process parentage issue preventing successful restart upon nfs-ganesha crash

samuelmanzer committed:
fix lint

samuelmanzer committed:
"glide up" nfs tests to remove warning

Commit:
nfs: improve error reporting for multiple service endpoints

Signed-off-by: Michal Minář <miminar@redhat.com>

aka-bo committed:
nfs-provisioner - added maxExports flag

The `maxExports` flag controls the maximum number of volumes a provisioner
is allowed to export. Once this limit is reached new claims will be
ignored with the hope that another provisioner will provision it.

Omitting this flag (or setting to a negative value) will disable this
feature, effectively allowing for unlimited exports.

spiffxp committed:
Rename OWNERS assignees: to approvers:

They are effectively the same, assignees is deprecated

hzxuzhonghu committed:
explicitly use a versioned core client

afbjorklund committed:
Use the official patch, rather than local copy

afbjorklund committed:
Upgrade container software versions slightly

Steffen911 committed:
missing space in readme

Steffen911 committed:
typo

Commit:
Merge pull request #544 from Steffen911/patch-1

missing space in readme

Commit:
Check strconv.ParseUint error.

`strconv.ParseUint` returns 0 and an error if argument is invalid. If we
don't check error, the following `chgrd` will succeed to change the
group of file to 0, and reports no error if the process is
running as root user. This is not expected.

aka-bo committed:
Expose `shouldProvision` to provisioners via new controller interface

Added new `controller.Qualifier` interface which is implemented by
any controller that has a `ShouldProvision` method. When implemented,
the provisioner's `ShouldProvision` method is called by lib controller's
`shouldProvision`, with a return value of false from the former
resulting in the same from the latter.

This allows provisioners the opportunity to ignore individual claims
prior to attaining leadership for the claim, avoiding uneccessary
election cycles when multiple provisioners are serving the same storage
class.

childsb committed:
Merge pull request #588 from aka-bo/nfs-provisioner-v1.0.9

Expose `shouldProvision` to provisioners via new controller interface

lawrencedudley committed:
update kubernetes api versions to be current to 1.8.8

lawrencedudley committed:
Merge branch 'master' of https://github.com/kubernetes-incubator/external-storage

initcron committed:
- updated version from v1alpha1 to v1 , else it breaks atleast on k8s 1.9.3
- added documentation to update namespace. does not work in custom namespaces otherwise

Commit:
API version moved from alpha to v1.

ianchakeres committed:
Fixed nfs-provisioner Makefile, so that it builds on osx.

Modified the build command and removed `go install` from build target. Added `go install` to `test-e2e` target.

It appears that hg (Mercurial) may be an e2e test dependency that isn't being checked or installed.

To rectify this issue and enable tests to pass on osx, I modified the test target to perform only unit tests. I created a new target test-all to execute both the unit tests and e2e tests.

weherdh committed:
Update image tag to latest v1.0.9 for easy testing

Commit:
Merge pull request #633 from schoolofdevops/master

- updated version from v1alpha1 to v1 , else it breaks atleast on k8s…

Commit:
Change all clusterroles to have endpoints permissions and reduced events permissions, consolidate where possible

krzwalko committed:
Add export-subnet for limit NFS clients

krzwalko committed:
Add warning when use-ganesha is true and export-subnet is specified

krzwalko committed:
Adjust test cases

Commit:
Revamp nfs e2e tests to fork k8s instead of vendoring it and use yamls

Commit:
Merge pull request #892 from wongma7/leader-election

Replace per-PVC leader election with per-cluster

Commit:
Fix nfs deployment.yaml

Commit:
Patch ganesha to skip over files when parsing mount table

Commit:
Add back openshift SCC

Commit:
Update to nfs-ganesha V2.7-rc2

Commit:
Try to prevent ss Update failure

Commit:
Distribute Role+Rolebinding everywhere instead of giving cluster-scoped endpoints r/w

Commit:
Fill in rbac.yaml with default namespace

bertinatto committed:
Remove typo in secret

Commit:
kc to kubectl in sample shell

Commit:
Merge pull request #984 from wongma7/kc

kc to kubectl in sample shell

Commit:
Copy docker's mount lib code instead of vendoring all of docker

Commit:
Merge pull request #986 from wongma7/1.12WIP

Update repo to use k8s.io/* 1.12

Commit:
Bump golang version to 1.11.1

Commit:
Override mountOptions parameter with sc.mountOptions

Commit:
Use kubernetes-sigs/sig-storage-lib-external-provisioner

Commit:
Fix verify errors, e.g. go 1.11 gofmt

Commit:
Add bit about nfs 1-replica limitation

liangxia committed:
Fix typo in file name

joaocc committed:
update nfs-ganesha to 2.7.1

k8s-ci-robot committed:
Merge pull request #1082 from hsz-devops/nfs-provisioner-v2.2.1-k8s1.12

Update nfs-ganesha to 2.7.1

hkolvenbach committed:
clarified nfs provisioner README.md regarding RBAC authorization

Aisuko committed:
Add serviceaccount of nfs-provisioner was created succeed logs to readme

uzuku committed:
[nfs-provisioner] Add error check for endpoints list when validating server service

cryptk committed:
Add ability to build nfs-provisioner for ARM

Commit:
Update nfs to lib 4 and client-go w GetReference fix

Mr-Linus committed:
Fix link failed.

These links have been deleted in commit <https://github.com/kubernetes-incubator/external-storage/commit/d46083d75be3c046c5313c03c0b4b5f29d9b2ec2#diff-4c2f31d7b36fe0ef8abda7acb20f1d86>

Mr-Linus committed:
Update README.md

thirdeyenick committed:
make the fsid device setting configurable

thirdeyenick committed:
improve description of the provisioner flag

thirdeyenick committed:
some improvements to the Dockerfile

* update to most current stable Ganesha version
* use a multistage build with a fedora minimal image

thirdeyenick committed:
add licensing information

wongma7 committed:
Fix NFS e2e to build using build/run.sh

k8s-ci-robot committed:
Merge pull request #1214 from thirdeyenick/nfs_provisioner_fsid

make the fsid device setting configurable

thirdeyenick committed:
ldconfig needs to be run after libraries have been copied

k8s-ci-robot committed:
Merge pull request #1215 from thirdeyenick/improve_dockerfile

Improve dockerfile and update Ganesha version

kvaps committed:
Add missing nfs ports and NFSv3 Support

Signed-off-by: kvaps <kvapss@gmail.com>

kvaps committed:
set NFS_V4_RECOV_ROOT to /export

k8s-ci-robot committed:
Merge pull request #1190 from uzuku/nfs-provision-error-check

[nfs-provisioner] Add error check for endpoints list when validating …

Commit:
Don't change existing behavior of nfs Makefile, just add ARM

kvaps committed:
add: setNlmPort function

kvaps committed:
fix failing test

k8s-ci-robot committed:
Merge pull request #1241 from kvaps/nfs-ports

Add missing ports and NFSv3 Support

wongma7 committed:
Fix cmake version for CHECK_SYMBOL_EXISTS

Commit:
Merge branch 'master' of /home/kiran_mova_mayadata_io/temp/migrate/external-storage into init-code

Commit:
update github references to new repo

Signed-off-by: kmova <kiran.mova@mayadata.io>

Commit:
update links/text to new repo in README.md

Signed-off-by: kmova <kiran.mova@mayadata.io>

ashishranjan738 commented on deploy/kubernetes/deployment.yaml:
```diff
@@ -0,0 +1,123 @@
+      serviceAccount: nfs-provisioner
+      containers:
+        - name: nfs-provisioner
+          image: quay.io/kubernetes_incubator/nfs-provisioner:latest
```
this needs to be changed right ?

ashishranjan738 commented on deploy/kubernetes/pod.yaml:
```diff
@@ -0,0 +1,64 @@
+  serviceAccount: nfs-provisioner
+  containers:
+    - name: nfs-provisioner
+      image: quay.io/kubernetes_incubator/nfs-provisioner:latest
```
same here

ashishranjan738 commented on deploy/kubernetes/statefulset.yaml:
```diff
@@ -0,0 +1,123 @@
+      terminationGracePeriodSeconds: 10
+      containers:
+        - name: nfs-provisioner
+          image: quay.io/kubernetes_incubator/nfs-provisioner:latest
```
same here

Commit:
update with new quay.io location

Update Makefile, examples and docs with the new location
for the NFS Ganesha Server and Provisioner image.

`quay.io/kubernetes_incubator/nfs-provisioner:latest` is
changed to `quay.io/external_storage/nfs-ganesha-server-and-provisioner:latest`

Signed-off-by: kmova <kiran.mova@mayadata.io>

Commit:
add dependencies via go mod vendor

Signed-off-by: kmova <kiran.mova@mayadata.io>

kmova commented:
@jsafrane @msau42 @wongma7 - this is ready for review. 

Also, @msau42 - could you help with removing the check of invalid commit similar to https://github.com/kubernetes-sigs/gluster-block-external-provisioner/pull/1#issuecomment-635445925

wongma7 commented:
IMO it's not necessary to continue using the quay external_storage org. Can move to docker hub, not use an org at all, or whatever makes sense.

msau42 commented:
Now we have community infra supported by wg-k8s-infra. I would reach out to them on how to get started. You can also take a look at the kubernetes-csi repos for examples on how to configure the builds. I think we can do that as a second step after getting the initial commits merged.

msau42 commented:
I've removed the invalid commit message label. I'm not sure how to handle the invalid CLA from the imported commits. @dims do you know?

kmova commented:
Took the help of linuxfoundation support and found that the issue is with the following older commits, where the github profile doesn't exist now. 
- 69c239b662f4ca15019b2df26766f0931f33a4d6
- 99301c73eaf8c9f3e349d2d82f5c286f7b70377a

msau42 commented:
Yeah, in the past we have force merged them but I'm not sure of the actual mechanics of how to do that. Is it just having an administrator hit the "Merge pull request" button?

dims commented:
@kmova please open a migration issue in https://github.com/kubernetes/org/issues/new/choose and ping the folks on the github admin team https://github.com/kubernetes/community/tree/master/github-management#github-administration-team and they will be able get the changes in.

kmova commented:
This PR had an issue with CLA which got resolved and merged via https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/pull/3
